x-pack/legacy/plugins/maps/public/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -107,8 +107,14 @@ export class GeojsonFileSource extends AbstractVectorSource {
   }
 
   async getGeoJsonWithMeta() {
+    const trimmedFeatures = this._descriptor.featureCollection.features
+      .map(({ type, geometry, properties = {} }) =>
+        ({ type, geometry, properties }));
```

Comment:
? I'm not sure how this is fixing the issue. This just copies a reference to the original properties. The properties are modified downstream. Which is why cloneDeep was introduced. But that cloning introduced a known performance hit.

```
const safeFeatures = this._descriptor.featureCollection.features.map(feature => {
return  {
   type: feature.type,//copy the value
   geometry: feature.geometry,//copy the reference. this avoid the performance hit that we'd otherwise get from deep-clone on large geometries (e.g. large polygons with thousands of nested coordinates, like army base file)
   properties: feature.properties ? { ...feature.properties } : {} //copy the properties. This is why cloneDeep was introduced in the first place
});
```

You could do the object destructuring for succinctness. But it is very dense and might be obscuring rather than clarifying.

Reply:
I like the destructuring on the properties you've done here. Originally on the file upload side, I destructured one level higher at the parsed geojson level and discovered it was still passing refs to the original nested properties object. Since we're only ever adding our own custom properties at the top level of the copied properties object, this should work well for us